UKHomeOffice/sgmr-service
# NMSW-34 Refactor pleasure craft form edit identification & API calls

## AC

Prepare pleasure craft form page for the new autocomplete component
Refactor how we identify it is an edit or add
Refactor API calls
Refactor when we store session data

----

## To test

- view pleasure craft page
- click add new pleasure craft
- fill in form
- save
- > it should save your craft
- click an existing pleasure craft update link
- change details
- save
- > it should save your changes
- click an existing pleasure craft update link
- change details
- click exit without saving
- click the same pleasure craft update link
- > it should not have saved your changes
- click delete & confirm
- > it should delete your pleasure craft

----

## Notes

I've moved and renamed the vessel page for consistency with rest of page naming/structure
